src/plugins/opensearch_dashboards_react/public/exit_full_screen_button/exit_full_screen_button.test.tsx

```diff
@@ -29,36 +29,79 @@
 import { ExitFullScreenButton } from './exit_full_screen_button';
 import { keys } from '@elastic/eui';
-import { mount } from 'enzyme';
+import { shallow } from 'enzyme';
+import { getLogosMock } from '../../../../core/common/mocks';
 
-test('is rendered', () => {
-  const component = mount(<ExitFullScreenButton onExitFullScreenMode={() => {}} />);
-
-  expect(component).toMatchSnapshot();
+const mockProps = () => ({
+  onExitFullScreenMode: jest.fn(),
+  logos: getLogosMock.default,
 });
 
-describe('onExitFullScreenMode', () => {
-  test('is called when the button is pressed', () => {
-    const onExitHandler = sinon.stub();
+describe('ExitFullScreenButton', () => {
+  it("is rendered using the dark theme's mark by default", () => {
+    const props = {
+      ...mockProps(),
+    };
+    const component = shallow(<ExitFullScreenButton {...props} />);
+    // In light color-scheme, the button has a dark background
+    const icons = component.find(`EuiIcon[type="${props.logos.Mark.dark.url}"]`);
 
-    const component = mount(<ExitFullScreenButton onExitFullScreenMode={onExitHandler} />);
+    expect(icons.length).toEqual(1);
 
-    component.find('button').simulate('click');
+    expect(component).toMatchSnapshot();
+  });
 
-    sinon.assert.calledOnce(onExitHandler);
+  it("is rendered using the dark theme's mark when light color-scheme is applied", () => {
+    const props = {
+      ...mockProps(),
+      logos: { ...getLogosMock.default, colorScheme: 'light' as const },
+    };
+    const component = shallow(<ExitFullScreenButton {...props} />);
+    // In light color-scheme, the button has a dark background
+    const icons = component.find(`EuiIcon[type="${props.logos.Mark.dark.url}"]`);
+
+    expect(icons.length).toEqual(1);
+
+    expect(component).toMatchSnapshot();
   });
 
-  test('is called when the ESC key is pressed', () => {
-    const onExitHandler = sinon.stub();
+  it("is rendered using the light theme's mark when dark color-scheme is applied", () => {
+    const props = {
+      ...mockProps(),
+      logos: { ...getLogosMock.default, colorScheme: 'dark' as const },
+    };
+    const component = shallow(<ExitFullScreenButton {...props} />);
+    // In light color-scheme, the button has a dark background
+    const icons = component.find(`EuiIcon[type="${props.logos.Mark.light.url}"]`);
+
+    expect(icons.length).toEqual(1);
+
+    expect(component).toMatchSnapshot();
+  });
+});
+
+describe('onExitFullScreenMode', () => {
+  it('is called when the button is pressed', () => {
+    const props = {
+      ...mockProps(),
+    };
+    const component = shallow(<ExitFullScreenButton {...props} />);
+    component.find('button').simulate('click');
+
+    expect(props.onExitFullScreenMode).toHaveBeenCalledTimes(1);
+  });
 
-    mount(<ExitFullScreenButton onExitFullScreenMode={onExitHandler} />);
+  it('is called when the ESC key is pressed', () => {
+    const props = {
+      ...mockProps(),
+    };
```

Comment:
Unnecessary memory io?

```suggestion
    const props = mockProps();
```